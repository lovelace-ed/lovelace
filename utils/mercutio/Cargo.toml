[package]
name = "mercutio"
version = "0.1.0"
authors = ["teymour-aldridge <teymour.aldridge@icloud.com>"]
edition = "2018"

[dependencies]
mercutio_codegen = { path = "../mercutio_codegen" }
malvolio = { path = "../malvolio" }
