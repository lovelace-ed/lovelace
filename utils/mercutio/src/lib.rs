pub use mercutio_codegen::CSS;

/// A trait which applies the relevant CSS styles to an item.
///
/// You can (but probably don't want to) implement this yourself, but we suggest using the derive
/// macro we provide.
///
/// ```rust
/// # use mercutio::*;
/// #[derive(CSS)]
/// #[font_family = "sans-serif"]
/// #[font_size = "24px"]
/// #[elements(H1, H2, H3)]
/// struct Title;
/// ```
pub trait Apply<T> {
    fn apply(t: T) -> T;
}
