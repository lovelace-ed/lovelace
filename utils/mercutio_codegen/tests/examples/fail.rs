use mercutio::*;

#[derive(CSS, Debug)]
#[font_family = "sans-serif"]
#[font_size = "24px"]
pub struct TitleStyles;

fn main() {

}
