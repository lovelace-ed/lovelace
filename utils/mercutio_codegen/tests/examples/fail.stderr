error: This item should have an attribute called `elements` (in the form `#[elements(H1, H2, P)])`, but that attribute does not exist on this element.Æ’
 --> $DIR/fail.rs:6:12
  |
6 | pub struct TitleStyles;
  |            ^^^^^^^^^^^
